src/engine/renderer/Material.cpp

```diff
@@ -0,0 +1,1946 @@
+	material->enableGridLighting = enableGridLighting;
+	material->enableGridDeluxeMapping = enableGridDeluxeMapping;
+	material->hasHeightMapInNormalMap = pStage->hasHeightMapInNormalMap;
+	material->enableReliefMapping = pStage->enableReliefMapping;
+	material->enableNormalMapping = pStage->enableNormalMapping && tr.cubeHashTable != nullptr;
+	material->enablePhysicalMapping = pStage->enablePhysicalMapping;
+	material->deformIndex = pStage->deformIndex;
+
+	gl_lightMappingShaderMaterial->SetDeluxeMapping( enableDeluxeMapping );
+
+	gl_lightMappingShaderMaterial->SetGridLighting( enableGridLighting );
+
+	gl_lightMappingShaderMaterial->SetGridDeluxeMapping( enableGridDeluxeMapping );
+
+	gl_lightMappingShaderMaterial->SetHeightMapInNormalMap( pStage->hasHeightMapInNormalMap );
+
+	gl_lightMappingShaderMaterial->SetReliefMapping( pStage->enableReliefMapping );
+
+	gl_lightMappingShaderMaterial->SetReflectiveSpecular( pStage->enableNormalMapping && tr.cubeHashTable != nullptr );
+
+	gl_lightMappingShaderMaterial->SetPhysicalShading( pStage->enablePhysicalMapping );
+
+	material->program = gl_lightMappingShaderMaterial->GetProgram( pStage->deformIndex );
+}
+
+static void ProcessMaterialReflection( Material* material, shaderStage_t* pStage ) {
+	material->shader = gl_reflectionShaderMaterial;
+
+	material->hasHeightMapInNormalMap = pStage->hasHeightMapInNormalMap;
+	material->enableReliefMapping = pStage->enableReliefMapping;
+	material->vertexAnimation = false;
+	material->deformIndex = pStage->deformIndex;
+
+	gl_reflectionShaderMaterial->SetHeightMapInNormalMap( pStage->hasHeightMapInNormalMap );
+
+	gl_reflectionShaderMaterial->SetReliefMapping( pStage->enableReliefMapping );
+
+	// gl_reflectionShaderMaterial->SetVertexSkinning( glConfig2.vboVertexSkinningAvailable && false );
+	gl_reflectionShaderMaterial->SetVertexAnimation( glState.vertexAttribsInterpolation > 0 && false );
+
+	material->program = gl_reflectionShaderMaterial->GetProgram( pStage->deformIndex );
+}
+
+static void ProcessMaterialSkybox( Material* material, shaderStage_t* pStage ) {
+	material->shader = gl_skyboxShaderMaterial;
+
+	material->deformIndex = pStage->deformIndex;
+
+	material->program = gl_skyboxShaderMaterial->GetProgram( pStage->deformIndex );
+}
+
+static void ProcessMaterialScreen( Material* material, shaderStage_t* pStage ) {
+	material->shader = gl_screenShaderMaterial;
+
+	material->deformIndex = pStage->deformIndex;
+
+	material->program = gl_screenShaderMaterial->GetProgram( pStage->deformIndex );
+}
+
+static void ProcessMaterialHeatHaze( Material* material, shaderStage_t* pStage, shader_t* shader ) {
+	material->shader = gl_heatHazeShaderMaterial;
+
+	material->vertexAnimation = false;
+	material->deformIndex = pStage->deformIndex;
+
+	// gl_heatHazeShaderMaterial->SetVertexSkinning( glConfig2.vboVertexSkinningAvailable && false );
+	gl_heatHazeShaderMaterial->SetVertexAnimation( glState.vertexAttribsInterpolation > 0 && false );
```

Comment:
Vertex/bone animation not supported currently and since this only draws BSP surfaces currently it's also not needed right now. I'll probably remove those lines altogether since I intend to use the same vertex shaders for surfaces both with and without animation (whenever I add support for that).

Reply:
Removed all the `&& false` and commented out stuff related to vertex/bone animation.